modules/ztp-creating-pgt-crs-with-hub-cluster-templates.adoc

```diff
@@ -0,0 +1,161 @@
+// Module included in the following assemblies:
```

Comment:
@imiller0 do you think we can add template use example in the official doc or in a knowledge based article?

Reply:
@jgato had some ideas. Perhaps we could generalize the example @jgato was working with a customer on?

Reply:
We would like to optimize some site policies using templat'ing. But I would need some time to analyze the existing ones, and which one would be enhanced with templating. One example, which would be pretty generic: to add extra labels/roles to different nodes. There is much manual-work-copy-paste if we do these just with Policies.

Reply:
Noting that with shut down fast approaching, we don't have a lot of time left to get this in. Is what we have OK for this release, or do we need to put our heads together to create deeper examples and use cases. @Missxiaoguo @imiller0 WDYT?

Reply:
Here is one example where I would put the SriovNetwork into the "group" PGT and then use hub side templating to specify the vlan (for example)

The entry in the group PolicyGenTemplate:
```
    - fileName: SriovNetwork.yaml
      policyName: "config-policy"
      metadata:
        name: "sriov-nw-du-mh"
        annotations:
          ran.openshift.io/ztp-deploy-wave: "10"
      spec:
        resourceName: du_mh
        vlan: '{{hub fromConfigMap "" "site-data" (printf "%s-vlan" .ManagedClusterName) | toInt hub}}'
```
And a configmap (also placed in git and synchronized to the hub cluster) which provides the vlan for site-1 and site-2 (as examples, one entry is needed per site). Note that this configmap must be in the same namespace with the policy that contains the template being substituted. The argocd annotation is needed only if this configmap will grow to a large size (~1MB) in which case it will fail to sync to the hub cluster due to the "lastApplied" annotation that gets added.
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: site-data
  namespace: ztp-group
  annotations:
    argocd.argoproj.io/sync-options: Replace=true
data:
  site-1-vlan: "101"
  site-2-vlan: "234"
```

Reply:
I've added this as an example that closely resembles the other update. PTAL and let me know if this works. 

Two use cases now: 
* [Specifying host NICs in site PolicyGenTemplate CRs with hub cluster templates](https://53209--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.html#ztp-specifying-nics-in-pgt-crs-with-hub-cluster-templates_ztp-advanced-policy-config)
* [Specifying VLAN IDs in group PolicyGenTemplate CRs with hub cluster templates](https://53209--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.html#ztp-managing-sriov-vlan-with-hub-cluster-templates-in-pgt_ztp-advanced-policy-config)